lib/backend/etcdbk/etcd.go

```diff
@@ -265,6 +265,7 @@ func New(ctx context.Context, params backend.Params, opts ...Option) (*EtcdBacke
 		CleanupInterval: utils.SeventhJitter(time.Minute * 2),
 	})
 	if err != nil {
+		cancel()
```

Comment:
If we're canceling the context here, shouldn't we also cancel it at other points in the initialization process?